### PR TITLE
Upload ONT data to DDS

### PR DESCRIPTION
**Pull request by ssjunnebo**
* Get project info from order portal
* Create a delivery project with DDS
* Upload the staged data to DDS

### CODE REVIEW

File: taca/delivery/cli.py [hunk 1]
```diff
@@ -33,22 +38,79 @@ def stage(project, flowcells, samples):
...
-    "-s",
     "--stage_dir",
     type=str,
     required=True,
     help="Staged directory to upload",
 )
+@click.option(
+    "--order_portal",
+    default=None,
+    envvar="ORDER_PORTAL",
```
[the remark this answers is not in the document]
Comment (alneberg) [reply taca/delivery/cli.py:52]: @ssjunnebo do you know which has precedence though if you would use the cli argument as well?

[the remark this answers is not in the document]
Comment (ssjunnebo) [reply taca/delivery/cli.py:52]: The cli option will take precedence

File: taca/delivery/cli.py [hunk 2]
```diff
@@ -33,22 +38,79 @@ def stage(project, flowcells, samples):
...
+    required=True,
+    type=click.File("r"),
+    help="Path to order portal credantials to retrive project information",
```
Comment (aanil) [inline taca/delivery/cli.py:55]:
```suggestion
    help="Path to order portal credentials to retrieve project information",
```

File: taca/delivery/delivery_classes.py [hunk 3]
```diff
@@ -111,3 +137,251 @@ def stage_data(self):
...
+                lab_email = self.order_details.get("fields", {}).get(
+                    "project_lab_email"
+                )
+                if (
+                    lab_email
+                    and lab_email != self.pi_email
+                    and lab_email not in other_member_details
+                ):
+                    other_member_details.append(lab_email)
+            except (AssertionError, ValueError):
```
Comment (aanil) [inline taca/delivery/delivery_classes.py:243]: Where is this exception thrown? What happens if the exception is thrown on bioinfo_email but it also has lab_email?

File: taca/delivery/cli.py (hunk 2 above)
Comment (ssjunnebo) [reply taca/delivery/cli.py]: Thanks!

File: taca/delivery/delivery_classes.py (hunk 3 above)
Comment (ssjunnebo) [reply taca/delivery/delivery_classes.py:243]: Good question... I copied this one over from taca-ngi-pipeline, I agree that it looks a bit unstable. I'll give it another look 👍

File: VERSIONLOG.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # TACA Version Log
 
+## 20241204.2
```
Comment (aanil) [inline VERSIONLOG.md:3]: Update the date maybe? There's a conflict with master

**Review by aanil — APPROVED** [review]
👍 Looks good! Let's see how it works out

File: taca/delivery/delivery_classes.py [hunk 5]
```diff
@@ -111,3 +137,251 @@ def stage_data(self):
...
+            if self.order_details.get("owner"):
+                owner_email = self.order_details.get("owner").get("email")
+                if (
+                    owner_email
+                    and owner_email != self.pi_email
+                    and owner_email not in other_member_details
+                ):
+                    other_member_details.append(owner_email)
+            if self.order_details.get("fields"):
+                bioinfo_email = self.order_details.get("fields").get("project_bx_email")
```
Comment (aanil) [inline taca/delivery/delivery_classes.py:226]: Maybe add a default, so it won't throw an exception if there is ever a case that `fields` doesn't exist?
```suggestion
                bioinfo_email = self.order_details.get("fields", {}).get("project_bx_email")
```

Comment (ssjunnebo) [reply taca/delivery/delivery_classes.py:226]: If `fields` doesn't exist, it won't enter the `if`. But yeah, I should add it to some other places I think.

**Pull request merged**